base/some.jl

```diff
@@ -145,3 +145,15 @@ macro something(args...)
     something = GlobalRef(Base, :something)
     return :($something($expr))
 end
+
+==(a::Some, b::Some)::Bool = a.value == b.value
+==(::Some{Missing}, ::Some) = false
+==(::Some, ::Some{Missing}) = false
+==(::Some{Missing}, ::Some{Missing}) = true
+
+isequal(a::Some, b::Some)::Bool = isequal(a.value, b.value)
+isequal(::Some{Missing}, ::Some) = false
+isequal(::Some, ::Some{Missing}) = false
+isequal(::Some{Missing}, ::Some{Missing}) = true
```

Comment:
Are these additional methods necessary? They are for `==` since that uses three value logic for `missing` but `isequal` does not, so I think simply forwarding to the wrapped values should cover these cases already (unless I'm missing something).

Reply:
I've added them in an effort to make the `test/missing.jl` tests happy & type stable, which checks that `isequal(::Any, ::Any)` always returns a `Bool`.

Reply:
I don't understand this either: why would these methods be needed given that `isequal` returns a `Bool` even for `missing`? And adding methods that are not needed to document something seems weird.

Reply:
Because of https://docs.julialang.org/en/v1/manual/performance-tips/#Be-aware-of-when-Julia-avoids-specializing.

```
==(a::Some, b::Some) = a.value == b.value
```

This doesn't specialize on the types of `a` or `b`, leading to the test failures seen earlier in CI.